src/WebUI/Services/TranslationManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Ninject;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebUI.Services
+{
+    class TranslationManager
+    {
+        private static volatile TranslationManager manager;
+        private static object syncRoot = new Object();
+
+        
+        public static TranslationManager Instance
+        {
+            get
+            {
+                if (manager == null)
+                {
+                    lock (syncRoot)
+                    {
+                        if (manager == null)
+                            manager = new TranslationManager();
+                    }
+                }
+
+                return manager;
+            }
+        }
+
+        private TranslationManager()
+        {
+           
+        }
+
+        public ITranslationProvider TranslationProvider { get; set; }
+       
+
+        public IEnumerable<CultureInfo> Languages
+        {
+            get
+            {
+                return TranslationProvider != null ? TranslationProvider.Languages : Enumerable.Empty<CultureInfo>();
+            }
+        }
+
+        private CultureInfo currentLanguage;
```

Comment:
The initial idea behind using `CultureInfo` was to use this with `Thread`'s culture in desktop apps. Later it was adopted for use in ASP.NET and to refrain from using Adapter or Facade I left everything as is. So, it would probably be easier to just use `string`.